builtin/providers/aws/resource_aws_waf_ipset.go

```diff
@@ -0,0 +1,175 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/waf"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsWafIPSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafIPSetCreate,
+		Read:   resourceAwsWafIPSetRead,
+		Update: resourceAwsWafIPSetUpdate,
+		Delete: resourceAwsWafIPSetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"ip_set_descriptors": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"value": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsWafIPSetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafconn
+
+	// ChangeToken
+	var ct *waf.GetChangeTokenInput
+	time.Sleep(10 * time.Second) // To Prevent WAFStaleDataException
+	res, err := conn.GetChangeToken(ct)
+	if err != nil {
+		return fmt.Errorf("Error getting change token: %s", err)
+	}
+
+	params := &waf.CreateIPSetInput{
+		ChangeToken: res.ChangeToken,
+		Name:        aws.String(d.Get("name").(string)),
+	}
+
+	resp, err := conn.CreateIPSet(params)
+	if err != nil {
+		return err
+	}
+	d.SetId(*resp.IPSet.IPSetId)
```

Comment:
If there a chance that resp.IPSet can be nil? I think it would be worth checking for the nil response here to prevent any potential pointer deference errors